voidcrew/code/game/objects/items/devices/walkman/walkman.dm

```diff
@@ -0,0 +1,340 @@
+#define sound_to(target, sound) target << (sound)
+#define NEXT_SONG_USE_TIMER (5 SECONDS)
+/obj/item/device/walkman
+	name = "walkman"
+	desc = "A cassette player that first hit the market over 200 years ago. Crazy how these never went out of style. Alt-click removes the Cassette. Ctrl-click changes to the next song"
+	icon = 'voidcrew/icons/obj/walkman.dmi'
+	icon_state = "walkman"
+	w_class = WEIGHT_CLASS_SMALL
+	actions_types = list(/datum/action/item_action/walkman/play_pause,/datum/action/item_action/walkman/next_song,/datum/action/item_action/walkman/restart_song)
+	///the cassette tape object
+	var/obj/item/device/cassette_tape/tape
+	///if the walkman is paused or not
+	var/paused = TRUE
+	///songs inside the current playlist
+	var/list/current_playlist = list()
+	///names of the songs inside the current playlist
+	var/list/current_songnames = list()
+	///Current song being played
+	var/sound/current_song
+	///Who's using the walkman
+	var/mob/current_listener
+	///Client of the listener
+	var/client/listener
+	///where in the playlist you are
+	var/pl_index = 1
+	///volume the walkman starts at
+	var/volume = 25
+	/// What kind of walkman design style to use
+	var/design = 1
+	///Is the current song a link? We handle those different
+	var/link_play = FALSE
+	///cooldown used by the next song to stop overlapping sounds between url based songs and normal ones
+	COOLDOWN_DECLARE(next_song_use)
+
+/obj/item/device/walkman/Initialize()
+	. = ..()
+	design = rand(1, 5)
+	update_icon()
+
+/obj/item/device/walkman/Destroy()
+	QDEL_NULL(tape)
+	break_sound()
+	current_song = null
+	current_listener = null
+	listener = null
+	STOP_PROCESSING(SSobj, src)
+	. = ..()
+
+/obj/item/device/walkman/attackby(obj/item/cassette, mob/user)
+	if(!istype(cassette, /obj/item/device/cassette_tape))
+		return
+	if(!tape)
+		insert_tape(cassette)
+		playsound(src,'sound/weapons/handcuffs.ogg',20,1)
+		to_chat(user,("You insert \the [cassette] into \the [src]"))
+	else
+		to_chat(user,("Remove the other tape first!"))
+
+/obj/item/device/walkman/attack_self(mob/user)
+	..()
+
+	if(!current_listener)
+		current_listener = user
+		listener = current_listener.client
+		START_PROCESSING(SSobj, src)
+	if(istype(tape))
+		if(paused)
+			play()
+			to_chat(user,("You press [src]'s 'play' button"))
+		else
+			pause()
+			to_chat(user,("You pause [src]"))
+		update_icon()
+	else
+		to_chat(user,("There's no tape to play"))
+	playsound(src,'sound/machines/click.ogg',20,1)
+
+/obj/item/device/walkman/AltClick(mob/user)
+	if(tape)
+		eject_tape(user)
+	else
+		..()
+
+/obj/item/device/walkman/CtrlClick(mob/user)
+	if(tape)
+		next_song(user)
+
```

Comment:
you should add an else return ..() here bc you cant drag empty walkmen using ctrl click

just a QoL fix